#Análisis de Personas Políticamente Expuestas (PEP) - Uruguay

**Autor:** Federico Ramos  
**Fecha:** Diciembre 2025  
**Formación:** Analista de Datos Jr. - Fundación Telefónica/UCU  
**Objetivo:** Análisis exploratorio para prevención de lavado de activos según Ley 19.574



# 📦 PASO 1: SETUP Y CARGA DE DATOS


In [1]:
# Importar librerías
import pandas as pd
import numpy as np

# Verificar que funcionan
print("✅ Pandas versión:", pd.__version__)
print("✅ Numpy versión:", np.__version__)
print("\n¡Todo listo para empezar!")

✅ Pandas versión: 2.2.2
✅ Numpy versión: 2.0.2

¡Todo listo para empezar!


In [2]:
# Configurar Pandas para mostrar números sin notación científica
pd.options.display.float_format = '{:.2f}'.format

print("✅ Pandas configurado para mostrar números normales")

✅ Pandas configurado para mostrar números normales


## 📂 1.1 - Carga del Dataset


In [3]:
# Subir archivo (va a aparecer un botón para seleccionar)
from google.colab import files
uploaded = files.upload()

# Cargar el CSV
df = pd.read_csv('lista-actualizada-de-pep.csv')
print(f"Dataset cargado: {df.shape}")


Saving lista-actualizada-de-pep.csv to lista-actualizada-de-pep.csv
Dataset cargado: (857, 4)


# 🔍 PASO 2: EXPLORACIÓN INICIAL

## 2.1 - Primeras Filas del Dataset


In [4]:
# Ver las primeras 5 filas
print("=" * 60)
print("PRIMERAS 5 FILAS DEL DATASET")
print("=" * 60)
df.head()


PRIMERAS 5 FILAS DEL DATASET


CI                                    NOMBRE  \
0  19995767                 ABAL BARU FERNANDO MARTIN   
1  14472184                  ABAL ORGUET ALICIA ELENA   
2  38735055                  ABAL RODRIGUEZ EVA MARIA   
3  31168574                 ABAL SABAÑO VICTOR MANUEL   
4  13274529  ABDALA REMERCIARI MANUEL JUAN WASHINGTON   

                                           CARGO  \
0                          ENCARGADO DE NEGOCIOS   
1                           DIRECTOR DE DIVISION   
2                                           EDIL   
3  DIRECTOR DE CONVIVENCIA Y SEGURIDAD CIUDADANA   
4                       REPRESENTANTE PERMANENTE   

                                           ORGANISMO  
0        MINISTERIO DE RELACIONES EXTERIORES (MRREE)  
1  INSTITUTO DEL NIÑO Y DEL ADOLESCENTE DEL URUGU...  
2                   JUNTA DEPARTAMENTAL DE MALDONADO  
3                       MINISTERIO DEL INTERIOR (MI)  
4        MINISTERIO DE RELACIONES EXTERIORES (MRREE)

## 📋 2.2 - Información General


In [5]:
# Información del dataset
print("\n" + "=" * 60)
print("INFORMACIÓN GENERAL")
print("=" * 60)
df.info()



INFORMACIÓN GENERAL
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 857 entries, 0 to 856
Data columns (total 4 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   CI         857 non-null    int64 
 1   NOMBRE     857 non-null    object
 2   CARGO      857 non-null    object
 3   ORGANISMO  857 non-null    object
dtypes: int64(1), object(3)
memory usage: 26.9+ KB


## 🔢 2.3 - Valores Únicos


In [6]:
print("=" * 60)
print("VALORES ÚNICOS POR COLUMNA")
print("=" * 60)

print(f"CI:        {df['CI'].nunique()} valores únicos")
print(f"NOMBRE:    {df['NOMBRE'].nunique()} valores únicos")
print(f"CARGO:     {df['CARGO'].nunique()} valores únicos")
print(f"ORGANISMO: {df['ORGANISMO'].nunique()} valores únicos")


VALORES ÚNICOS POR COLUMNA
CI:        857 valores únicos
NOMBRE:    856 valores únicos
CARGO:     179 valores únicos
ORGANISMO: 108 valores únicos


## 📊 2.4 - Estadísticas Descriptivas


In [7]:
# Estadísticas descriptivas
print("\n" + "=" * 60)
print("ESTADÍSTICAS DESCRIPTIVAS")
print("=" * 60)
df.describe()



ESTADÍSTICAS DESCRIPTIVAS


CI
count      857.00
mean  29850095.75
std   11198865.21
min    8861335.00
25%   18596013.00
50%   30975291.00
75%   38795746.00
max   56396655.00

# 🧹 PASO 3: LIMPIEZA Y VERIFICACIÓN

## 🔍 3.1 - Verificación de Duplicados


In [8]:
print("=" * 60)
print("VERIFICACIÓN DE DUPLICADOS")
print("=" * 60)

# ¿Hay filas completamente duplicadas?
duplicados_totales = df.duplicated().sum()
print(f"\nFilas duplicadas (completas): {duplicados_totales}")

if duplicados_totales > 0:
    print("⚠️  Hay duplicados que eliminar")
else:
    print("✅ No hay filas duplicadas")


VERIFICACIÓN DE DUPLICADOS

Filas duplicadas (completas): 0
✅ No hay filas duplicadas


## 🆔 3.2 - Verificación de CIs Duplicadas


In [9]:
# ¿Hay CIs repetidas? (esto sería raro, cada persona tiene 1 CI)
duplicados_ci = df['CI'].duplicated().sum()
print(f"\nCIs duplicadas: {duplicados_ci}")

if duplicados_ci > 0:
    print("⚠️  Algunas personas aparecen más de una vez")
    print("\nMostrando personas con CI duplicada:")
    print(df[df['CI'].duplicated(keep=False)].sort_values('CI'))
else:
    print("✅ Cada CI aparece solo una vez")



CIs duplicadas: 0
✅ Cada CI aparece solo una vez


## 👥 3.3 - Nombres Duplicados (Homónimos)


In [10]:
# ¿Quiénes tienen el mismo nombre?
duplicados_nombre = df['NOMBRE'].duplicated().sum()
print(f"\nNombres duplicados: {duplicados_nombre}")

if duplicados_nombre > 0:
    print("\nPersonas con el mismo nombre:")
    nombres_repetidos = df[df['NOMBRE'].duplicated(keep=False)].sort_values('NOMBRE')
    print(nombres_repetidos[['CI', 'NOMBRE', 'CARGO']])



Nombres duplicados: 1

Personas con el mismo nombre:
           CI                     NOMBRE                     CARGO
719  13994911  BOTTA LUQUIN DANIEL JULIO        CONSUL DE DISTRITO
720  12776996  BOTTA LUQUIN DANIEL JULIO  JEFE DE SECCION CONSULAR


## ✅ 3.4 - Resumen de Calidad


In [11]:
print("\n" + "=" * 60)
print("RESUMEN DE CALIDAD DE DATOS")
print("=" * 60)

print(f"✅ Total de registros: {len(df)}")
print(f"✅ Valores nulos: {df.isnull().sum().sum()}")
print(f"✅ Filas duplicadas: {df.duplicated().sum()}")
print(f"✅ CIs duplicadas: {df['CI'].duplicated().sum()}")
print("\n🎯 Conclusión: El dataset está limpio y listo para analizar")



RESUMEN DE CALIDAD DE DATOS
✅ Total de registros: 857
✅ Valores nulos: 0
✅ Filas duplicadas: 0
✅ CIs duplicadas: 0

🎯 Conclusión: El dataset está limpio y listo para analizar


# 📊 PASO 4: ANÁLISIS EXPLORATORIO

## 🏛️ 4.1 - Top 10 Organismos con mas PePs


In [12]:
print("=" * 60)
print("TOP 10 ORGANISMOS CON MÁS PEPs")
print("=" * 60)

# Contar cuántas personas hay por organismo
top_organismos = df['ORGANISMO'].value_counts().head(10)

# Mostrar con porcentajes
for i, (organismo, cantidad) in enumerate(top_organismos.items(), 1):
    porcentaje = (cantidad / len(df)) * 100
    print(f"{i:2}. {organismo[:45]:45} → {cantidad:3} ({porcentaje:4.1f}%)")


TOP 10 ORGANISMOS CON MÁS PEPs
 1. PODER JUDICIAL                                →  97 (11.3%)
 2. PODER LEGISLATIVO                             →  66 ( 7.7%)
 3. FISCALIA GENERAL DE LA NACION                 →  59 ( 6.9%)
 4. MINISTERIO DE RELACIONES EXTERIORES (MRREE)   →  44 ( 5.1%)
 5. MINISTERIO DEL INTERIOR (MI)                  →  35 ( 4.1%)
 6. JUNTA DEPARTAMENTAL DE CANELONES              →  31 ( 3.6%)
 7. MINISTERIO DE ECONOMIA Y FINANZAS (MEF)       →  31 ( 3.6%)
 8. INSTITUTO DEL NIÑO Y DEL ADOLESCENTE DEL URUG →  22 ( 2.6%)
 9. JUNTA DEPARTAMENTAL DE PAYSANDU               →  20 ( 2.3%)
10. MINISTERIO DE SALUD PUBLICA (MSP)             →  20 ( 2.3%)


## 🏛️ 4.2 - Top 10 cargos mas comunes de PePs

In [13]:
print("\n" + "=" * 60)
print("TOP 10 CARGOS MÁS COMUNES")
print("=" * 60)

top_cargos = df['CARGO'].value_counts().head(10)

for i, (cargo, cantidad) in enumerate(top_cargos.items(), 1):
    porcentaje = (cantidad / len(df)) * 100
    print(f"{i:2}. {cargo:30} → {cantidad:3} ({porcentaje:4.1f}%)")



TOP 10 CARGOS MÁS COMUNES
 1. EDIL                           → 171 (20.0%)
 2. JUEZ LETRADO                   →  65 ( 7.6%)
 3. DIRECTOR                       →  42 ( 4.9%)
 4. REPRESENTANTE NACIONAL         →  35 ( 4.1%)
 5. FISCAL LETRADA ADSCRIPTA       →  27 ( 3.2%)
 6. GERENTE                        →  25 ( 2.9%)
 7. JUEZ DE PAZ                    →  20 ( 2.3%)
 8. GERENTE DE AREA                →  19 ( 2.2%)
 9. DIRECTOR DE DIVISION           →  18 ( 2.1%)
10. DIRECTOR GENERAL               →  17 ( 2.0%)


## 🎯 4.3 - Categorización por Nivel de Riesgo (v1)


In [14]:
print("=" * 60)
print("CREANDO CATEGORÍA: NIVEL DE RIESGO")
print("=" * 60)

# Función para categorizar cargos
def categorizar_cargo(cargo):
    """
    Esta función lee el nombre del cargo y devuelve el nivel de riesgo
    """
    cargo_mayuscula = cargo.upper()  # Convertir a mayúsculas para buscar

    # Alto Riesgo - Poder Ejecutivo/Legislativo
    if any(palabra in cargo_mayuscula for palabra in
           ['MINISTRO', 'SUBSECRETARIO', 'PRESIDENTE', 'VICEPRESIDENTE',
            'INTENDENTE', 'EMBAJADOR', 'SENADOR', 'REPRESENTANTE NACIONAL']):
        return 'Alto - Ejecutivo/Legislativo'

    # Alto Riesgo - Poder Judicial
    elif any(palabra in cargo_mayuscula for palabra in
             ['JUEZ', 'FISCAL', 'MAGISTRADO']):
        return 'Alto - Poder Judicial'

    # Alto Riesgo - Dirección General
    elif any(palabra in cargo_mayuscula for palabra in
             ['DIRECTOR GENERAL', 'DIRECTOR NACIONAL', 'GERENTE GENERAL']):
        return 'Alto - Dirección General'

    # Medio Riesgo - Dirección/Gerencia
    elif any(palabra in cargo_mayuscula for palabra in
             ['DIRECTOR', 'GERENTE', 'SUBDIRECTOR', 'COORDINADOR']):
        return 'Medio - Dirección/Gerencia'

    # Medio Riesgo - Representación Local
    elif any(palabra in cargo_mayuscula for palabra in
             ['EDIL', 'CONCEJAL', 'ALCALDE']):
        return 'Medio - Representación Local'

    # Otros
    else:
        return 'Otros'

# Aplicar la función a todas las filas
df['NIVEL_RIESGO'] = df['CARGO'].apply(categorizar_cargo)

print("✅ Nueva columna 'NIVEL_RIESGO' creada\n")

# Ver cuántas personas hay en cada nivel
print("Distribución por Nivel de Riesgo:")
print("-" * 60)
niveles = df['NIVEL_RIESGO'].value_counts()

for nivel, cantidad in niveles.items():
    porcentaje = (cantidad / len(df)) * 100
    print(f"{nivel:35} → {cantidad:3} ({porcentaje:5.1f}%)")


CREANDO CATEGORÍA: NIVEL DE RIESGO
✅ Nueva columna 'NIVEL_RIESGO' creada

Distribución por Nivel de Riesgo:
------------------------------------------------------------
Medio - Dirección/Gerencia          → 246 ( 28.7%)
Medio - Representación Local        → 186 ( 21.7%)
Alto - Poder Judicial               → 144 ( 16.8%)
Otros                               → 126 ( 14.7%)
Alto - Ejecutivo/Legislativo        → 114 ( 13.3%)
Alto - Dirección General            →  41 (  4.8%)


## ✔️ 4.4 - Verificación de Categorización


In [15]:
print("=" * 60)
print("VERIFICACIÓN: ¿La categorización funcionó bien?")
print("=" * 60)

# Ver ejemplos de cada categoría
print("\n1. Ejemplos de ALTO RIESGO - Ejecutivo/Legislativo:")
print(df[df['NIVEL_RIESGO'] == 'Alto - Ejecutivo/Legislativo'][['NOMBRE', 'CARGO']].head(3))

print("\n2. Ejemplos de ALTO RIESGO - Poder Judicial:")
print(df[df['NIVEL_RIESGO'] == 'Alto - Poder Judicial'][['NOMBRE', 'CARGO']].head(3))

print("\n3. Ejemplos de MEDIO RIESGO - Representación Local:")
print(df[df['NIVEL_RIESGO'] == 'Medio - Representación Local'][['NOMBRE', 'CARGO']].head(3))

print("\n4. Ejemplos de OTROS:")
print(df[df['NIVEL_RIESGO'] == 'Otros'][['NOMBRE', 'CARGO']].head(3))


VERIFICACIÓN: ¿La categorización funcionó bien?

1. Ejemplos de ALTO RIESGO - Ejecutivo/Legislativo:
                              NOMBRE           CARGO
5        ABDALA SCHWARZ PABLO DANIEL   SUBSECRETARIO
11   ABILLEIRA ALVAREZ DARIO AURELIO  VICEPRESIDENTE
14  ABRACINSKAS CEPLIKAS ROSA LILIAN        SENADORA

2. Ejemplos de ALTO RIESGO - Poder Judicial:
                             NOMBRE           CARGO
6        ABDALA SOSA MERCEDES MARIA  FISCAL LETRADO
10  ABERASTEGUI PALMINA ANA CLAUDIA    JUEZ LETRADO
15  ABRAHAM BANCHERO MARIA VICTORIA    JUEZ LETRADO

3. Ejemplos de MEDIO RIESGO - Representación Local:
                               NOMBRE     CARGO
2            ABAL RODRIGUEZ EVA MARIA      EDIL
16  ABRAHAM LEONE GUSTAVO MAXIMILIANO      EDIL
26   ACERENZA SAN JUAN FRANCISCO JOSE  CONCEJAL

4. Ejemplos de OTROS:
                                      NOMBRE  \
0                  ABAL BARU FERNANDO MARTIN   
4   ABDALA REMERCIARI MANUEL JUAN WASHINGTON   
29                   

## 🏢 4.5 - Categorización por Tipo de Organismo

In [16]:
print("=" * 60)
print("CREANDO CATEGORÍA: TIPO DE ORGANISMO")
print("=" * 60)

def categorizar_organismo(organismo):
    """
    Clasifica el organismo según su naturaleza
    """
    org_mayuscula = organismo.upper()

    if 'MINISTERIO' in org_mayuscula:
        return 'Ministerios'

    elif 'PODER JUDICIAL' in org_mayuscula or 'FISCALIA' in org_mayuscula or 'TRIBUNAL' in org_mayuscula:
        return 'Poder Judicial'

    elif 'PODER LEGISLATIVO' in org_mayuscula or 'JUNTA DEPARTAMENTAL' in org_mayuscula:
        return 'Poder Legislativo'

    elif 'INTENDENCIA' in org_mayuscula:
        return 'Gobiernos Departamentales'

    elif 'BANCO' in org_mayuscula or 'BPS' in org_mayuscula or 'BSE' in org_mayuscula:
        return 'Instituciones Financieras'

    elif any(empresa in org_mayuscula for empresa in ['ANTEL', 'ANCAP', 'OSE', 'UTE', 'ANP']):
        return 'Empresas Públicas'

    elif 'UNIVERSIDAD' in org_mayuscula or 'ANEP' in org_mayuscula:
        return 'Educación'

    else:
        return 'Otros Organismos'

# Aplicar la función
df['TIPO_ORGANISMO'] = df['ORGANISMO'].apply(categorizar_organismo)

print("✅ Nueva columna 'TIPO_ORGANISMO' creada\n")

# Ver distribución
print("Distribución por Tipo de Organismo:")
print("-" * 60)
tipos = df['TIPO_ORGANISMO'].value_counts()

for tipo, cantidad in tipos.items():
    porcentaje = (cantidad / len(df)) * 100
    print(f"{tipo:30} → {cantidad:3} ({porcentaje:5.1f}%)")


CREANDO CATEGORÍA: TIPO DE ORGANISMO
✅ Nueva columna 'TIPO_ORGANISMO' creada

Distribución por Tipo de Organismo:
------------------------------------------------------------
Poder Legislativo              → 244 ( 28.5%)
Ministerios                    → 172 ( 20.1%)
Poder Judicial                 → 158 ( 18.4%)
Otros Organismos               →  98 ( 11.4%)
Gobiernos Departamentales      →  87 ( 10.2%)
Instituciones Financieras      →  52 (  6.1%)
Empresas Públicas              →  35 (  4.1%)
Educación                      →  11 (  1.3%)


## 🔍 4.6 - Investigación: Errores Detectados


In [17]:
print("=" * 60)
print("INVESTIGANDO: ¿Qué pasó con Carolina Ache?")
print("=" * 60)

# Buscar por nombre
carolina = df[df['NOMBRE'].str.contains('ACHE', case=False)]
print(carolina[['NOMBRE', 'CARGO', 'NIVEL_RIESGO']])


INVESTIGANDO: ¿Qué pasó con Carolina Ache?
                         NOMBRE                                   CARGO  \
29         ACHE BATLLE CAROLINA  SUBSECRETARIA DE RELACIONES EXTERIORES   
723  BOTTARO IRACHE CELIA MARIA                                    EDIL   

                     NIVEL_RIESGO  
29                          Otros  
723  Medio - Representación Local  


## 📋 4.7 - Revisión de Casos "Otros"


In [18]:
print("\n" + "=" * 60)
print("TODOS LOS CARGOS CLASIFICADOS COMO 'OTROS'")
print("=" * 60)

otros = df[df['NIVEL_RIESGO'] == 'Otros'][['NOMBRE', 'CARGO', 'ORGANISMO']]
print(f"\nTotal de 'Otros': {len(otros)}\n")

# Mostrar los primeros 20 para revisar
print("Primeros 20 casos:")
print(otros.head(20).to_string())



TODOS LOS CARGOS CLASIFICADOS COMO 'OTROS'

Total de 'Otros': 126

Primeros 20 casos:
                                       NOMBRE                                   CARGO                                                                             ORGANISMO
0                   ABAL BARU FERNANDO MARTIN                   ENCARGADO DE NEGOCIOS                                           MINISTERIO DE RELACIONES EXTERIORES (MRREE)
4    ABDALA REMERCIARI MANUEL JUAN WASHINGTON                REPRESENTANTE PERMANENTE                                           MINISTERIO DE RELACIONES EXTERIORES (MRREE)
29                       ACHE BATLLE CAROLINA  SUBSECRETARIA DE RELACIONES EXTERIORES                                           MINISTERIO DE RELACIONES EXTERIORES (MRREE)
30                       ACHUGAR DIAZ MARIANA                               CONSEJERA                                                  UNIVERSIDAD DE LA REPUBLICA (UDELAR)
58                 AGRASAR VIGNOLO OMAR MARIO        

## 🔧 4.8 - Categorización Mejorada (v2)

**Iteración:** crear → verificar → mejorar

In [19]:
print("=" * 60)
print("MEJORANDO LA CATEGORIZACIÓN")
print("=" * 60)

def categorizar_cargo_v2(cargo):
    """
    Versión mejorada con más casos
    """
    cargo_mayuscula = cargo.upper()

    # Alto Riesgo - Poder Ejecutivo/Legislativo
    # IMPORTANTE: Agregamos versiones femeninas y más cargos
    if any(palabra in cargo_mayuscula for palabra in
           ['MINISTRO', 'MINISTRA', 'SUBSECRETARIO', 'SUBSECRETARIA',
            'PRESIDENTE', 'PRESIDENTA', 'VICEPRESIDENTE', 'VICEPRESIDENTA',
            'INTENDENTE', 'INTENDENTA', 'EMBAJADOR', 'EMBAJADORA',
            'SENADOR', 'SENADORA', 'REPRESENTANTE NACIONAL',
            'REPRESENTANTE PERMANENTE', 'GENERAL', 'ALMIRANTE']):
        return 'Alto - Ejecutivo/Legislativo'

    # Alto Riesgo - Poder Judicial
    elif any(palabra in cargo_mayuscula for palabra in
             ['JUEZ', 'JUEZA', 'FISCAL', 'MAGISTRADO', 'MAGISTRADA']):
        return 'Alto - Poder Judicial'

    # Alto Riesgo - Dirección General
    elif any(palabra in cargo_mayuscula for palabra in
             ['DIRECTOR GENERAL', 'DIRECTORA GENERAL',
              'DIRECTOR NACIONAL', 'DIRECTORA NACIONAL',
              'GERENTE GENERAL']):
        return 'Alto - Dirección General'

    # Medio-Alto - Cargos diplomáticos y policiales
    elif any(palabra in cargo_mayuscula for palabra in
             ['CONSUL GENERAL', 'CÓNSUL GENERAL',
              'JEFE DE POLICIA', 'SUBJEFE DE POLICIA',
              'ENCARGADO DE NEGOCIOS', 'COMISARIO GENERAL']):
        return 'Medio-Alto - Diplomático/Policial'

    # Medio Riesgo - Dirección/Gerencia
    elif any(palabra in cargo_mayuscula for palabra in
             ['DIRECTOR', 'DIRECTORA', 'GERENTE',
              'SUBDIRECTOR', 'SUBDIRECTORA', 'COORDINADOR', 'COORDINADORA']):
        return 'Medio - Dirección/Gerencia'

    # Medio Riesgo - Representación Local
    elif any(palabra in cargo_mayuscula for palabra in
             ['EDIL', 'CONCEJAL', 'ALCALDE', 'ALCALDESA']):
        return 'Medio - Representación Local'

    # Otros
    else:
        return 'Otros'

# Aplicar la nueva función
df['NIVEL_RIESGO'] = df['CARGO'].apply(categorizar_cargo_v2)

print("✅ Categorización actualizada\n")

# Ver cómo cambió
print("Nueva distribución:")
print("-" * 60)
niveles = df['NIVEL_RIESGO'].value_counts()

for nivel, cantidad in niveles.items():
    porcentaje = (cantidad / len(df)) * 100
    print(f"{nivel:35} → {cantidad:3} ({porcentaje:5.1f}%)")

print("\n¿Cuántos 'Otros' quedan ahora?")
print(f"→ {len(df[df['NIVEL_RIESGO'] == 'Otros'])} personas")


MEJORANDO LA CATEGORIZACIÓN
✅ Categorización actualizada

Nueva distribución:
------------------------------------------------------------
Medio - Dirección/Gerencia          → 229 ( 26.7%)
Medio - Representación Local        → 186 ( 21.7%)
Alto - Ejecutivo/Legislativo        → 181 ( 21.1%)
Alto - Poder Judicial               → 143 ( 16.7%)
Otros                               →  87 ( 10.2%)
Alto - Dirección General            →  20 (  2.3%)
Medio-Alto - Diplomático/Policial   →  11 (  1.3%)

¿Cuántos 'Otros' quedan ahora?
→ 87 personas


## ✅ 4.9 - Verificación de Correcciones


In [20]:
# Verificar que Carolina Ache cambió
print("\n" + "=" * 60)
print("VERIFICACIÓN: ¿Se corrigió Carolina Ache?")
print("=" * 60)

carolina = df[df['NOMBRE'].str.contains('ACHE', case=False)]
print(carolina[['NOMBRE', 'CARGO', 'NIVEL_RIESGO']])



VERIFICACIÓN: ¿Se corrigió Carolina Ache?
                         NOMBRE                                   CARGO  \
29         ACHE BATLLE CAROLINA  SUBSECRETARIA DE RELACIONES EXTERIORES   
723  BOTTARO IRACHE CELIA MARIA                                    EDIL   

                     NIVEL_RIESGO  
29   Alto - Ejecutivo/Legislativo  
723  Medio - Representación Local  


## 📅 4.10 - Estimación de Rango Etario


In [21]:
print("=" * 60)
print("CREANDO CATEGORÍA: RANGO DE EDAD (ESTIMADO)")
print("=" * 60)

def estimar_edad(ci):
    """
    Estima rango de edad según la CI uruguaya
    CIs más bajas = personas mayores
    """
    if ci < 20000000:
        return 'Mayor 50 años'
    elif ci < 40000000:
        return '35-50 años'
    else:
        return 'Menor 35 años'

# Aplicar la función
df['RANGO_EDAD'] = df['CI'].apply(estimar_edad)

print("✅ Nueva columna 'RANGO_EDAD' creada\n")

# Ver distribución
print("Distribución por Rango de Edad:")
print("-" * 60)
edades = df['RANGO_EDAD'].value_counts().sort_index()

for edad, cantidad in edades.items():
    porcentaje = (cantidad / len(df)) * 100
    print(f"{edad:15} → {cantidad:3} ({porcentaje:5.1f}%)")


CREANDO CATEGORÍA: RANGO DE EDAD (ESTIMADO)
✅ Nueva columna 'RANGO_EDAD' creada

Distribución por Rango de Edad:
------------------------------------------------------------
35-50 años      → 412 ( 48.1%)
Mayor 50 años   → 260 ( 30.3%)
Menor 35 años   → 185 ( 21.6%)


In [22]:
print("\n" + "=" * 60)
print("EJEMPLOS: CI vs EDAD ESTIMADA")
print("=" * 60)

# Mostrar algunos casos para ver si tiene sentido
ejemplos = df[['CI', 'NOMBRE', 'RANGO_EDAD']].sample(10).sort_values('CI')
print(ejemplos.to_string(index=False))



EJEMPLOS: CI vs EDAD ESTIMADA
      CI                               NOMBRE    RANGO_EDAD
10333685         ABREU BONILLA SERGIO ENRIQUE Mayor 50 años
11839505            BARTOLOTTA DE LEON RAFAEL Mayor 50 años
12776996            BOTTA LUQUIN DANIEL JULIO Mayor 50 años
12893754 BARRERA MALRECHAUFFE GRACIELA HAYDEE Mayor 50 años
13496884           BAYARDI LOZANO JOSE ARTURO Mayor 50 años
30309824           BORBA SIERRA SILVIA SUSANA    35-50 años
35206198               ARTOLA ROCHA JUAN LUIS    35-50 años
37117979        BUONO CANEPA SABRINA CRISTINA    35-50 años
38409460             BELLOMO PERAZA MARIA SOL    35-50 años
39207265         ALMADA MALLADA PABLO ALBERTO    35-50 años


---
# 🔥 PASO 5: ANÁLISIS CRUZADO

## 🚨 5.1 - Alto Riesgo por Organismo


In [23]:
print("=" * 60)
print("ANÁLISIS CRUZADO 1: Alto Riesgo por Organismo")
print("=" * 60)

# Filtrar solo Alto Riesgo
alto_riesgo = df[df['NIVEL_RIESGO'].str.contains('Alto')]

# Top 10 organismos con más PEPs de Alto Riesgo
top_alto = alto_riesgo['ORGANISMO'].value_counts().head(10)

print(f"\nTotal de PEPs de Alto Riesgo: {len(alto_riesgo)} ({len(alto_riesgo)/len(df)*100:.1f}%)\n")
print("Top 10 organismos con PEPs de Alto Riesgo:")
print("-" * 60)

for i, (org, cant) in enumerate(top_alto.items(), 1):
    porcentaje = (cant / len(alto_riesgo)) * 100
    print(f"{i:2}. {org[:50]:50} {cant:3} ({porcentaje:4.1f}%)")


ANÁLISIS CRUZADO 1: Alto Riesgo por Organismo

Total de PEPs de Alto Riesgo: 355 (41.4%)

Top 10 organismos con PEPs de Alto Riesgo:
------------------------------------------------------------
 1. PODER JUDICIAL                                      94 (26.5%)
 2. PODER LEGISLATIVO                                   63 (17.7%)
 3. FISCALIA GENERAL DE LA NACION                       56 (15.8%)
 4. MINISTERIO DE RELACIONES EXTERIORES (MRREE)         25 ( 7.0%)
 5. MINISTERIO DEL INTERIOR (MI)                        17 ( 4.8%)
 6. MINISTERIO DE ECONOMIA Y FINANZAS (MEF)              7 ( 2.0%)
 7. MINISTERIO DE SALUD PUBLICA (MSP)                    6 ( 1.7%)
 8. MINISTERIO DE DEFENSA NACIONAL (MDN)                 5 ( 1.4%)
 9. INTENDENCIA DE CANELONES                             4 ( 1.1%)
10. MINISTERIO DE DESARROLLO SOCIAL (MIDES)              4 ( 1.1%)


## 🏦 5.2 - PEPs en Instituciones Financieras


In [24]:
print("\n" + "=" * 60)
print("ANÁLISIS CRUZADO 2: PEPs en Instituciones Financieras")
print("=" * 60)

# Filtrar instituciones financieras
fin = df[df['TIPO_ORGANISMO'] == 'Instituciones Financieras']

print(f"\nTotal de PEPs en instituciones financieras: {len(fin)}\n")

# ¿Qué nivel de riesgo tienen?
print("Distribución por Nivel de Riesgo:")
print("-" * 60)
for nivel, cant in fin['NIVEL_RIESGO'].value_counts().items():
    porcentaje = (cant / len(fin)) * 100
    print(f"{nivel:35} → {cant:2} ({porcentaje:5.1f}%)")

print("\n¿En qué instituciones están?")
print("-" * 60)
for inst, cant in fin['ORGANISMO'].value_counts().items():
    print(f"  {inst}: {cant}")



ANÁLISIS CRUZADO 2: PEPs en Instituciones Financieras

Total de PEPs en instituciones financieras: 52

Distribución por Nivel de Riesgo:
------------------------------------------------------------
Medio - Dirección/Gerencia          → 46 ( 88.5%)
Alto - Ejecutivo/Legislativo        →  4 (  7.7%)
Otros                               →  2 (  3.8%)

¿En qué instituciones están?
------------------------------------------------------------
  BANCO DE PREVISION SOCIAL (BPS): 19
  BANCO DE SEGUROS DEL ESTADO (BSE): 15
  BANCO DE LA REPUBLICA ORIENTAL DEL URUGUAY (BROU): 11
  BANCO CENTRAL DEL URUGUAY (BCU): 5
  BANCO HIPOTECARIO DEL URUGUAY (BHU): 2


In [25]:
print("\n" + "=" * 60)
print("ANÁLISIS CRUZADO 3: Edad de PEPs de Alto Riesgo")
print("=" * 60)

# Filtrar Alto Riesgo
alto_riesgo = df[df['NIVEL_RIESGO'].str.contains('Alto')]

print("Distribución etaria de PEPs de Alto Riesgo:")
print("-" * 60)

for edad, cant in alto_riesgo['RANGO_EDAD'].value_counts().sort_index().items():
    porcentaje = (cant / len(alto_riesgo)) * 100
    print(f"{edad:15} → {cant:3} ({porcentaje:5.1f}%)")



ANÁLISIS CRUZADO 3: Edad de PEPs de Alto Riesgo
Distribución etaria de PEPs de Alto Riesgo:
------------------------------------------------------------
35-50 años      → 168 ( 47.3%)
Mayor 50 años   → 111 ( 31.3%)
Menor 35 años   →  76 ( 21.4%)


In [26]:
print("\n" + "=" * 60)
print("TABLA RESUMEN EJECUTIVO")
print("=" * 60)

# Crear tabla cruzada: Tipo de Organismo vs Nivel de Riesgo
tabla_resumen = pd.crosstab(
    df['TIPO_ORGANISMO'],
    df['NIVEL_RIESGO'],
    margins=True,  # Agregar totales
    margins_name='TOTAL'
)

print("\nCantidad de PEPs por Tipo de Organismo y Nivel de Riesgo:")
print(tabla_resumen)

# Guardarlo en CSV para usar después
tabla_resumen.to_csv('tabla_cruzada_pep.csv')
print("\n✅ Tabla guardada en: tabla_cruzada_pep.csv")



TABLA RESUMEN EJECUTIVO

Cantidad de PEPs por Tipo de Organismo y Nivel de Riesgo:
NIVEL_RIESGO               Alto - Dirección General  \
TIPO_ORGANISMO                                        
Educación                                         0   
Empresas Públicas                                 0   
Gobiernos Departamentales                         0   
Instituciones Financieras                         0   
Ministerios                                      19   
Otros Organismos                                  0   
Poder Judicial                                    1   
Poder Legislativo                                 0   
TOTAL                                            20   

NIVEL_RIESGO               Alto - Ejecutivo/Legislativo  \
TIPO_ORGANISMO                                            
Educación                                             1   
Empresas Públicas                                     6   
Gobiernos Departamentales                            25   
Instituciones F

## 🚨 5.3 - CASOS CRÍTICOS DE: Alto Riesgo en Finanzas

**Máxima prioridad para Compliance**


In [27]:
print("=" * 60)
print("🚨 ANÁLISIS CRÍTICO: Alto Riesgo en Finanzas")
print("=" * 60)

# Filtrar: Alto Riesgo + Instituciones Financieras
criticos = df[
    (df['NIVEL_RIESGO'].str.contains('Alto')) &
    (df['TIPO_ORGANISMO'] == 'Instituciones Financieras')
]

print(f"\n⚠️  TOTAL DE CASOS CRÍTICOS: {len(criticos)}\n")
print("Estos son PEPs de Alto Riesgo que trabajan EN instituciones financieras.")
print("Requieren MÁXIMA diligencia según Ley 19.574\n")
print("=" * 60)

if len(criticos) > 0:
    # Mostrar todos los casos
    print("\nLISTA COMPLETA DE CASOS CRÍTICOS:")
    print("-" * 60)

    for idx, row in criticos.iterrows():
        print(f"\n👤 {row['NOMBRE']}")
        print(f"   📋 Cargo: {row['CARGO']}")
        print(f"   🏛️  Organismo: {row['ORGANISMO']}")
        print(f"   ⚠️  Nivel de Riesgo: {row['NIVEL_RIESGO']}")
        print(f"   📅 Rango Edad: {row['RANGO_EDAD']}")
        print(f"   🆔 CI: {row['CI']}")

    print("\n" + "=" * 60)
    print("💡 RECOMENDACIÓN:")
    print("   • Monitoreo reforzado de transacciones")
    print("   • Actualización trimestral de perfil")
    print("   • Verificación de conflictos de interés")
    print("   • Seguimiento de patrimonio declarado")

    # Guardar lista en CSV
    criticos[['CI', 'NOMBRE', 'CARGO', 'ORGANISMO', 'NIVEL_RIESGO']].to_csv(
        'pep_alto_riesgo_finanzas.csv',
        index=False
    )
    print("\n✅ Lista exportada: pep_alto_riesgo_finanzas.csv")

else:
    print("No se encontraron PEPs de Alto Riesgo en instituciones financieras.")


🚨 ANÁLISIS CRÍTICO: Alto Riesgo en Finanzas

⚠️  TOTAL DE CASOS CRÍTICOS: 4

Estos son PEPs de Alto Riesgo que trabajan EN instituciones financieras.
Requieren MÁXIMA diligencia según Ley 19.574


LISTA COMPLETA DE CASOS CRÍTICOS:
------------------------------------------------------------

👤 AMORIN BATLLE JOSE GERARDO
   📋 Cargo: PRESIDENTE
   🏛️  Organismo: BANCO DE SEGUROS DEL ESTADO (BSE)
   ⚠️  Nivel de Riesgo: Alto - Ejecutivo/Legislativo
   📅 Rango Edad: Mayor 50 años
   🆔 CI: 11676993

👤 BATALLA GARCIA DA ROSA ALFONSINA
   📋 Cargo: VICEPRESIDENTE
   🏛️  Organismo: BANCO DE SEGUROS DEL ESTADO (BSE)
   ⚠️  Nivel de Riesgo: Alto - Ejecutivo/Legislativo
   📅 Rango Edad: Menor 35 años
   🆔 CI: 42453065

👤 BORRELLI MARCHI ROBERTO
   📋 Cargo: SECRETARIO GENERAL
   🏛️  Organismo: BANCO DE LA REPUBLICA ORIENTAL DEL URUGUAY (BROU)
   ⚠️  Nivel de Riesgo: Alto - Ejecutivo/Legislativo
   📅 Rango Edad: Mayor 50 años
   🆔 CI: 13642291

👤 BURSTIN LEDERFAIN DARIO GASTON
   📋 Cargo: VICEPRESID

## 📈 5.4 - Análisis Detallado: Sector Financiero
## 📋 5.5 - Registro: Medio Riesgo en Finanzas



In [28]:
print("=" * 80)
print("📊 REGISTRO: OTROS PEPs EN INSTITUCIONES FINANCIERAS")
print("=" * 80)

fin_todas = df[df['TIPO_ORGANISMO'] == 'Instituciones Financieras']
# Define fin_resto: personas en instituciones financieras que NO son de 'Alto' riesgo
fin_resto = fin_todas[~fin_todas['NIVEL_RIESGO'].str.contains('Alto')]

# Niveles que quedan (sin los "Alto")
niveles_orden = [
    'Medio-Alto - Diplomático/Policial',
    'Medio - Dirección/Gerencia',
    'Medio - Representación Local',
    'Otros'
]

contador_general = 1

for nivel in niveles_orden:
    personas_nivel = fin_resto[fin_resto['NIVEL_RIESGO'] == nivel]

    if len(personas_nivel) == 0:
        continue

    print("\n" + "█" * 80)
    print(f"║ NIVEL DE RIESGO: {nivel}")
    print(f"║ Total: {len(personas_nivel)} persona(s)")
    print("█" * 80)

    # Ordenar por organismo y nombre
    personas_ordenado = personas_nivel.sort_values(['ORGANISMO', 'NOMBRE'])

    organismo_actual = None

    for idx, row in personas_ordenado.iterrows():
        # Mostrar organismo cuando cambia
        if row['ORGANISMO'] != organismo_actual:
            organismo_actual = row['ORGANISMO']
            print(f"\n┌─ {organismo_actual}")
            print("│")

        # Datos de la persona
        print(f"│  {contador_general:2}. 👤 {row['NOMBRE']}")
        print(f"│      📋 Cargo: {row['CARGO']}")
        print(f"│      🆔 CI: {row['CI']}")
        print(f"│      📅 Edad: {row['RANGO_EDAD']}")
        print("│")

        contador_general += 1

print("\n" + "=" * 80)
print(f"✅ TOTAL (excluyendo Alto Riesgo): {len(fin_resto)} PEPs")
print("=" * 80)

# Exportar solo los 48
columnas = ['CI', 'NOMBRE', 'CARGO', 'ORGANISMO', 'NIVEL_RIESGO', 'RANGO_EDAD']
export = fin_resto[columnas].copy()
export = export.sort_values(['NIVEL_RIESGO', 'ORGANISMO', 'NOMBRE'])
export.to_csv('REGISTRO_48_PEPs_MEDIO_RIESGO_FINANZAS.csv', index=False)

print("\n✅ Archivo: REGISTRO_48_PEPs_MEDIO_RIESGO_FINANZAS.csv")
print("   (Los 48 restantes, sin los 4 críticos)")

📊 REGISTRO: OTROS PEPs EN INSTITUCIONES FINANCIERAS

████████████████████████████████████████████████████████████████████████████████
║ NIVEL DE RIESGO: Medio - Dirección/Gerencia
║ Total: 46 persona(s)
████████████████████████████████████████████████████████████████████████████████

┌─ BANCO CENTRAL DEL URUGUAY (BCU)
│
│   1. 👤 AHARONIAN FRANZINI CRISTINA
│      📋 Cargo: GERENTE DE AREA
│      🆔 CI: 15263831
│      📅 Edad: Mayor 50 años
│
│   2. 👤 ALLO ARRIETA ALFREDO MIGUEL
│      📋 Cargo: GERENTE
│      🆔 CI: 17935789
│      📅 Edad: Mayor 50 años
│
│   3. 👤 ARTECONA GULLA DANIEL EDUARDO
│      📋 Cargo: GERENTE
│      🆔 CI: 17438957
│      📅 Edad: Mayor 50 años
│

┌─ BANCO DE LA REPUBLICA ORIENTAL DEL URUGUAY (BROU)
│
│   4. 👤 ABRAHAM RAMOS ADRIANA AURORA
│      📋 Cargo: GERENTE EJECUTIVO
│      🆔 CI: 18027646
│      📅 Edad: Mayor 50 años
│
│   5. 👤 ALBERTI POGGI JUAN ANTONIO
│      📋 Cargo: GERENTE EJECUTIVO
│      🆔 CI: 17611082
│      📅 Edad: Mayor 50 años
│
│   6. 👤 ALONSO MARIA 

---
# 📄 PASO 6: COMUNICACIÓN Y EXPORTACIÓN

## 📝 6.1 - Reporte Ejecutivo Final


In [29]:
print("=" * 70)
print(" " * 15 + "REPORTE EJECUTIVO FINAL")
print("=" * 70)

# Calcular estadísticas clave
alto_total = len(df[df['NIVEL_RIESGO'].str.contains('Alto')])
medio_total = len(df[df['NIVEL_RIESGO'].str.contains('Medio')])
fin_todas = df[df['TIPO_ORGANISMO'] == 'Instituciones Financieras']
criticos = df[(df['NIVEL_RIESGO'].str.contains('Alto')) &
              (df['TIPO_ORGANISMO'] == 'Instituciones Financieras')]
fin_resto = fin_todas[~fin_todas['NIVEL_RIESGO'].str.contains('Alto')]

resumen = f"""
╔══════════════════════════════════════════════════════════════════════╗
║        ANÁLISIS DE PERSONAS POLÍTICAMENTE EXPUESTAS (PEP)           ║
║                    URUGUAY - DICIEMBRE 2025                          ║
║                 Prevención de Lavado de Activos                      ║
╚══════════════════════════════════════════════════════════════════════╝

📊 DATOS GENERALES
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
• Total de PEPs registrados: {len(df):,}
• Organismos distintos: {df['ORGANISMO'].nunique()}
• Cargos distintos: {df['CARGO'].nunique()}
• Fuente: SENACLAFT - Catálogo de Datos Abiertos
• Marco legal: Ley N° 19.574 (Prevención de Lavado de Activos)

🎯 DISTRIBUCIÓN POR NIVEL DE RIESGO
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
"""

# Mostrar niveles ordenados
niveles_ordenados = [
    'Alto - Ejecutivo/Legislativo',
    'Alto - Poder Judicial',
    'Alto - Dirección General',
    'Medio-Alto - Diplomático/Policial',
    'Medio - Dirección/Gerencia',
    'Medio - Representación Local',
    'Otros'
]

for nivel in niveles_ordenados:
    cant = len(df[df['NIVEL_RIESGO'] == nivel])
    if cant > 0:
        pct = (cant/len(df)*100)
        resumen += f"• {nivel:40} {cant:3} ({pct:5.1f}%)\n"

resumen += f"\n  → TOTAL ALTO RIESGO: {alto_total} ({alto_total/len(df)*100:.1f}%)\n"
resumen += f"  → TOTAL MEDIO RIESGO: {medio_total} ({medio_total/len(df)*100:.1f}%)\n"

resumen += f"""
🏛️  DISTRIBUCIÓN POR TIPO DE ORGANISMO
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
"""

for tipo, cant in df['TIPO_ORGANISMO'].value_counts().items():
    pct = (cant/len(df)*100)
    resumen += f"• {tipo:30} {cant:3} ({pct:5.1f}%)\n"

resumen += f"""
👥 PERFIL ETARIO (Estimado por CI)
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
"""

for edad in ['Menor 35 años', '35-50 años', 'Mayor 50 años']:
    cant = len(df[df['RANGO_EDAD'] == edad])
    pct = (cant/len(df)*100)
    resumen += f"• {edad:20} {cant:3} ({pct:5.1f}%)\n"

resumen += f"""
🚨 CASOS CRÍTICOS: Alto Riesgo en Instituciones Financieras
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
• Total identificados: {len(criticos)}
• Requieren debida diligencia REFORZADA según Art. 8 Ley 19.574

Distribución por institución:
"""

for org in sorted(criticos['ORGANISMO'].unique()):
    cant = len(criticos[criticos['ORGANISMO'] == org])
    # Mostrar cargos específicos
    cargos = criticos[criticos['ORGANISMO'] == org]['CARGO'].tolist()
    org_corto = org.replace('BANCO DE LA REPUBLICA ORIENTAL DEL URUGUAY', 'BROU').replace('BANCO DE SEGUROS DEL ESTADO', 'BSE').replace('BANCO HIPOTECARIO DEL URUGUAY', 'BHU')
    resumen += f"  • {org_corto}: {cant} PEP(s)\n"
    for cargo in cargos:
        resumen += f"      - {cargo}\n"

resumen += f"""
📊 CONTEXTO: Instituciones Financieras (Todos los niveles)
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
• Total de PEPs en sector financiero: {len(fin_todas)} ({len(fin_todas)/len(df)*100:.1f}% del total)
• Alto Riesgo: {len(criticos)} ({len(criticos)/len(fin_todas)*100:.1f}% del sector)
• Medio Riesgo: {len(fin_resto)} ({len(fin_resto)/len(fin_todas)*100:.1f}% del sector)

Distribución por institución:
  • BPS (Previsión Social): {len(fin_todas[fin_todas['ORGANISMO'].str.contains('BPS')])} PEPs
  • BSE (Seguros): {len(fin_todas[fin_todas['ORGANISMO'].str.contains('BSE')])} PEPs
  • BROU (Banca Comercial): {len(fin_todas[fin_todas['ORGANISMO'].str.contains('BROU')])} PEPs
  • BCU (Banca Central): {len(fin_todas[fin_todas['ORGANISMO'].str.contains('BCU')])} PEPs
  • BHU (Hipotecario): {len(fin_todas[fin_todas['ORGANISMO'].str.contains('BHU')])} PEPs

📋 RECOMENDACIONES PARA SUJETOS OBLIGADOS
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
1. PRIORIZACIÓN DE RECURSOS
   • Foco inmediato: {len(criticos)} casos críticos (Alto Riesgo en finanzas)
   • Monitoreo reforzado: {alto_total} PEPs de Alto Riesgo total
   • Seguimiento estándar: {medio_total} PEPs de Medio Riesgo

2. MONITOREO CONTINUO
   • Actualización trimestral desde fuente oficial (catalogodatos.gub.uy)
   • Verificación automática por CI en proceso de onboarding
   • Cross-check con familiares directos (cónyuge, hijos, padres)
   • Alertas por cambios de cargo o institución

3. DEBIDA DILIGENCIA REFORZADA (Art. 8 Ley 19.574)
   Para PEPs de Alto Riesgo:
   • Aprobación de alta dirección para iniciar relación comercial
   • Identificación del origen de fondos y patrimonio
   • Monitoreo continuo de transacciones
   • Revisión anual del perfil de riesgo

   Especial atención a:
   • {len(df[df['NIVEL_RIESGO'] == 'Alto - Poder Judicial'])} PEPs en Poder Judicial
   • {len(df[df['NIVEL_RIESGO'] == 'Alto - Ejecutivo/Legislativo'])} PEPs Ejecutivo/Legislativo
   • {len(criticos)} PEPs de Alto Riesgo DENTRO de instituciones financieras

4. GESTIÓN DOCUMENTAL
   • Mantener expediente actualizado de cada PEP
   • Justificar por escrito decisiones sobre relaciones de alto riesgo
   • Registrar transacciones inusuales en sistema de alertas
   • Reporte mensual a Oficial de Cumplimiento
   • Archivo de evidencia para inspecciones SENACLAFT

5. CAPACITACIÓN DEL PERSONAL
   • Entrenamiento anual en identificación de PEPs
   • Actualización sobre cambios en lista oficial
   • Casos prácticos de señales de alerta
   • Procedimientos de escalamiento

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
📁 ARCHIVOS GENERADOS:
   • Dataset completo enriquecido con categorías de análisis
   • Lista de {len(criticos)} casos críticos para seguimiento inmediato
   • Registro de {len(fin_resto)} PEPs de medio riesgo en finanzas
   • Tablas resumen para integración con sistemas KYC/AML

💡 VALOR DEL ANÁLISIS:
   Este estudio permite a las instituciones financieras cumplir eficientemente
   con la normativa de prevención de lavado de activos, priorizando recursos
   en los casos de mayor exposición y automatizando la verificación de PEPs.

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Analista: Federico Ramos
Formación: Analista de Datos Jr. - Fundación Telefónica/UCU (300hs)
Fecha: {pd.Timestamp.now().strftime('%d de %B de %Y')}
Herramientas: Python 3.x, Pandas, Google Colab
Fuente datos: SENACLAFT - https://catalogodatos.gub.uy/
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
"""

print(resumen)

# Guardar en archivo de texto
with open('REPORTE_EJECUTIVO_PEP.txt', 'w', encoding='utf-8')  as f:
    f.write(resumen)

print("\n✅ Reporte guardado: REPORTE_EJECUTIVO_PEP.txt")
print("   Este archivo puede imprimirse o enviarse por email a stakeholders")


               REPORTE EJECUTIVO FINAL

╔══════════════════════════════════════════════════════════════════════╗
║        ANÁLISIS DE PERSONAS POLÍTICAMENTE EXPUESTAS (PEP)           ║
║                    URUGUAY - DICIEMBRE 2025                          ║
║                 Prevención de Lavado de Activos                      ║
╚══════════════════════════════════════════════════════════════════════╝

📊 DATOS GENERALES
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
• Total de PEPs registrados: 857
• Organismos distintos: 108
• Cargos distintos: 179
• Fuente: SENACLAFT - Catálogo de Datos Abiertos
• Marco legal: Ley N° 19.574 (Prevención de Lavado de Activos)

🎯 DISTRIBUCIÓN POR NIVEL DE RIESGO
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
• Alto - Ejecutivo/Legislativo             181 ( 21.1%)
• Alto - Poder Judicial                    143 ( 16.7%)
• Alto - Dirección General                  20 (  2.3%)
• Medio-Alto - Diplomático/Policia

---
# ✅ PROYECTO COMPLETADO

📊 **857 PEPs analizados y categorizados**  
🚨 **4 casos críticos identificados**  
📦 **Archivos exportados para uso operativo**

*Federico Ramos - Diciembre 2025*

In [30]:
import os

print("=" * 60)
print("📦 ARCHIVOS GENERADOS EN ESTE PROYECTO")
print("=" * 60)

archivos_csv = [f for f in os.listdir() if f.endswith('.csv')]
archivos_txt = [f for f in os.listdir() if f.endswith('.txt')]

print("\n📄 Archivos CSV:")
for archivo in sorted(archivos_csv):
    size = os.path.getsize(archivo) / 1024
    print(f"  • {archivo} ({size:.1f} KB)")

print("\n📝 Archivos TXT:")
for archivo in sorted(archivos_txt):
    size = os.path.getsize(archivo) / 1024
    print(f"  • {archivo} ({size:.1f} KB)")

print("\n✅ Proyecto completado exitosamente")


📦 ARCHIVOS GENERADOS EN ESTE PROYECTO

📄 Archivos CSV:
  • REGISTRO_48_PEPs_MEDIO_RIESGO_FINANZAS.csv (6.1 KB)
  • lista-actualizada-de-pep.csv (71.2 KB)
  • pep_alto_riesgo_finanzas.csv (0.5 KB)
  • tabla_cruzada_pep.csv (0.5 KB)

📝 Archivos TXT:
  • REPORTE_EJECUTIVO_PEP.txt (7.0 KB)

✅ Proyecto completado exitosamente
